### PR TITLE
doc: reconcile release checklist

### PR DESCRIPTION
**Pull request by batrick**
Updating the releases documentation existed already but that section should be moved to after dev freeze when the branch is forked.

Fixes: 7b5017913a9ecf27ca7281367a5472a9b9b043ec

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

File: doc/dev/release-checklists.rst [hunk 1]
```diff
@@ -145,7 +138,9 @@ After dev freeze
 ================
 
 - [ ] create branch for new release
-- [ ] remove release notes for release branch: see also 33d63c32cbf81ed81ce2185063011b98846a0d44
+- [ ] remove ``doc/releases/*.rst``. This should leave behind ``doc/releases/releases.yml`` which is used for doc building purposes.
```
Comment (idryomov) [inline doc/dev/release-checklists.rst:141]: I liked that `see also 33d63c32cbf81ed81ce2185063011b98846a0d44` -- there is a lot of additional detail in the commit message there.  I'd suggest appending
> See commit 33d63c32cbf81ed81ce2185063011b98846a0d44 ("doc: remove release notes for release branch") for details.

here.

File: doc/dev/release-checklists.rst [hunk 2]
```diff
@@ -145,7 +138,9 @@ After dev freeze
...
+- [ ] cherry-pick 8cf9ad62949516666ad0f2c0bb7726ef68e4d666 ("doc: add releases links to toc"). There will be trivial conflicts.
+- [ ] add redirect for new major release at `RTD <https://readthedocs.org/dashboard/ceph/redirects/>`_.
```
Comment (idryomov) [inline doc/dev/release-checklists.rst:143]: Just want to double check that it's indeed the case and that there is no way to do set it up just once with some sort of wild card?

File: doc/dev/release-checklists.rst (hunk 1 above)
Comment (batrick) [reply doc/dev/release-checklists.rst]: done, thanks

File: doc/dev/release-checklists.rst (hunk 2 above)
Comment (batrick) [reply doc/dev/release-checklists.rst:143]: I tried wildcards in the past and it didn't work. The configuration options are somewhat clunky and primitive.

**Review by idryomov — APPROVED** [review]

**Pull request merged**